src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/ILogger/DatadogLoggingScope.cs

```diff
@@ -0,0 +1,96 @@
+﻿// <copyright file="DatadogLoggingScope.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.ILogger
+{
+    internal class DatadogLoggingScope : IReadOnlyList<KeyValuePair<string, object>>
+    {
+        private readonly string _service;
+        private readonly string _env;
+        private readonly string _version;
+        private readonly Tracer _tracer;
+
+        public DatadogLoggingScope()
+            : this(Tracer.Instance)
+        {
+        }
+
+        internal DatadogLoggingScope(Tracer tracer)
+        {
+            _tracer = tracer;
+            _service = tracer.DefaultServiceName ?? string.Empty;
+            _env = tracer.Settings.Environment ?? string.Empty;
+            _version = tracer.Settings.ServiceVersion ?? string.Empty;
+        }
+
+        public int Count => 4;
+
+        public KeyValuePair<string, object> this[int index]
+        {
+            get
+            {
+                if (index == 0)
+                {
+                    return new KeyValuePair<string, object>("dd_service", _tracer.ActiveScope?.Span.ServiceName ?? _service);
+                }
+                else if (index == 1)
+                {
+                    return new KeyValuePair<string, object>("dd_env", _env);
+                }
+                else if (index == 2)
+                {
+                    return new KeyValuePair<string, object>("dd_version", _version);
+                }
+                else if (index == 3)
+                {
+                    return new KeyValuePair<string, object>("dd_trace_id", _tracer.ActiveScope?.Span.TraceId);
+                }
+
+                throw new ArgumentOutOfRangeException(nameof(index));
+            }
+        }
+
+        public override string ToString()
+        {
+            var span = _tracer.ActiveScope?.Span;
+            if (span is null)
+            {
+                return string.Empty;
+            }
+
+            return string.Format(
+                CultureInfo.InvariantCulture,
+                "dd_service:{0}, dd_env:{1}, dd_version:{2}, dd_trace_id:{3}",
+                span.ServiceName,
+                _env,
+                _version,
+                span.TraceId);
+        }
+
+        public IEnumerator<KeyValuePair<string, object>> GetEnumerator()
+        {
+            var span = _tracer.ActiveScope?.Span;
+            if (span is null)
+            {
+                yield break;
+            }
```

Comment:
I'm undecided whether we should still add the `dd_service`, `dd_env`, and `dd_version` tags if we don't have a trace. I'm inclined to say we should, but our other logs injection enrichers _don't_ do this currently, so I took this approach for consistency.

Does anyone else have opinions on which approach to take? Should we always add the `dd_service` etc tags?

Reply:
Established that we _should_ be doing this, so implemented in https://github.com/DataDog/dd-trace-dotnet/pull/1663/commits/0ba8588b8feb0b6106ea2d1c1e9b90e786ccd32e and updated the tests in https://github.com/DataDog/dd-trace-dotnet/pull/1663/commits/893deff858a845a23d06df32b845b63c9e5681bb